base/grisu/grisu.jl

```diff
@@ -116,7 +116,7 @@ end
 
 function Base.show(io::IO, x::Union{Float64,Float32})
     if get(io, :compact, false)
-        _show(io, x, PRECISION, 6, true, true)
+        _show(io, x, PRECISION, 6, isa(x,Float32), true)
```

Comment:
Sorry, I got this condition backwards, which is why the test is failing. Should be `x isa Float64`.